CIME/scripts/create_test.py

```
@@ -361,6 +365,15 @@ def parse_command_line(args, description):
         "\noption will allow existing baseline directories to be silently overwritten.",
     )
 
+    parser.add_argument(
+        "--allow-baseline-skip",
+        action="store_true",
+        default=default,
+        help="If the --generate option is given, then an attempt to overwrite "
+        "\nan existing baseline directory will raise an error. WARNING: Specifying this "
+        "\noption will allow tests with existing baseline directories to be silently skipped.",
```

Based on my understanding of the code, I'd suggest changing this documentation. My understanding is that (1) the test itself is rerun, it's just the baseline generation phase that's skipped; and (2) it's not really silently skipped, because (if I'm reading the code right) it will lead to a FAIL result in the baseline generation stage. If my understanding is right about these things, can you change the wording of the documentation to make this more clear? (I'd be happy to talk this over with you if it would help.)

---

Oops, good catch! The description is actually correct, but the code wasn't doing what I wanted. The idea is that I want tests with existing baselines to not even be set up, let alone run. Fixing this now (hopefully).

---

> The idea is that I want tests with existing baselines to not even be set up, let alone run.

I can see arguments for either behavior (i.e., either what it said in the documentation or how the code is currently working). If you do change the behavior to be what you intended, then I think the name of this argument should change as well: to me, anyway, "--allow-baseline-skip" implies something more like what the code is currently doing; if you entirely skip setting up and running a test because the baseline already exists, I think this needs a name that implies this, like "--skip-tests-with-existing-baselines" (or maybe something a little less wordy?). The difference, in my mind, is that starting this with "allow" somehow suggests to me that not much will change and it's just bypassing an error condition, whereas the behavior you say you want feels a lot more active.

---

Sure; changed to that with 3f28e481f. I think I'm fine with the wordiness, but maybe it'd be good to add a synonym like `-s`?

---

> Sure; changed to that with [3f28e48](https://github.com/ESMCI/cime/commit/3f28e481fd5a71abf43fe593e3c3acf88b336d95). I think I'm fine with the wordiness, but maybe it'd be good to add a synonym like `-s`?

Thanks! Given the potential for silently skipping tests, I lean towards requiring a long, explicitly-named option so that the user is sure to know what they're doing when setting the option (as opposed to potentially thinking that something like `-s` does one thing when in fact it does this other thing that could be problematic and hard to realize that it did it).

---

I agree with @billsacks that non-standard things that won't be used as often should NOT have a shorthand name (and -s is usually used for --silent, so would be confusing). I like the idea of having a more explicit name that accurately describes what's going on. In the initial proposal I like that the name was similar to the --allow-baseline-overwrite option, but with the error checking and the exclusive group and everything I don't think this is needed now.

So I like --skip-tests-with-existing-baselines as I think it's really clear on what it does. So this looks good to me other than possibly adding some testing around this.

@samsrabin do you want to look at adding tests together or would you like to work with @billsacks or someone else on that? I'd like to put some effort in there, even if we only do enough to decide it isn't worth it for this, right now.

Other than that I'm looking forward to seeing this get merged in!

---

Thanks, Erik! It sounds like there aren't any pre-existing tests that already exist for `create_test`, so I'd be starting from scratch. I think that's outside the scope of this PR and should probably be raised as a separate issue.

---

Nope, I'm wrong: As Bill mentioned, there is some system-testing of `create_test`. However, there isn't any testing that involves generating a baseline, so I think it'd still require more effort than I'm able to put in for now.

---

@samsrabin actually according to @billsacks comment here: https://github.com/ESMCI/cime/pull/4737#issuecomment-2599362106
there are a few that could possibly be added to. I haven't looked at them so I'm not sure how involved this would be though. And I'm totally fine if you want to open an issue and have that part done later.

I expect that we'll be using this with CTSM testing so likely will find bugs that crop up with it. In principle though because I do expect to use it, that's also a reason for doing some testing. But, in any case, I'm good either way here...

---

Ahh, OK. My message got sent before I saw your latest update. Go ahead and create an issue and we'll have this as a task for a later PR. It would be good for us in the CTSM team to have more experience in adding cime testing, and that's another reason to do this. But, again we'll do that later.